lib/spack/spack/test/stage.py

```diff
@@ -93,94 +168,220 @@ def get_stage_path(stage, stage_name):
         return stage.path
 
 
-@pytest.fixture()
-def tmpdir_for_stage(mock_archive):
-    """Uses a temporary directory for staging"""
-    current = spack.paths.stage_path
+@pytest.fixture
+def no_tmp_root_stage(monkeypatch):
+    """
+    Disable use of a temporary root for staging.
+
+    Note that it can be important for other tests that the previous settings be
+    restored when the test case is over.
+    """
+    current_root = spack.stage._tmp_root
+    current_use_tmp_stage = spack.stage._use_tmp_stage
+    monkeypatch.setattr(spack.stage, '_tmp_root', None)
+    monkeypatch.setattr(spack.stage, '_use_tmp_stage', False)
+    yield
+    monkeypatch.setattr(spack.stage, '_tmp_root', current_root)
+    monkeypatch.setattr(spack.stage, '_use_tmp_stage', current_use_tmp_stage)
```

Comment:
I *believe* that `monkeypatch` automatically reverts these (that's the point of `monkeypatch`), so you do not need to track the current values and restore them.  Can you see if you can simplify this?

Reply:
I didn't want to assume but that's an excellent point.  Will try.